doc/syntax.bnf

```diff
@@ -83,7 +83,7 @@ reserved "why3"
   | "@"? <qident>
   | "_"
   | "?" - <ident> {"[" <term> {"," <term>}* "]"}?
-  | "&" - <ident> {"[" <term> {"," <term>}* "]"}?
+  | "&" - {<ident> | "_"} {"[" <term> {"," <term>}* "]"}?
```

Comment:
Since <ident> includes "_", "&" - {<ident> | "_"} is equivalent to "&" - <ident>. Shouldn't you change the definition of <ident> so that it excludes "_"? Or any current ident starting with "_"?

Reply:
These are not equivalent, `<ident>` does not allow `"_"` since it is reserved.

Reply:
Ok. Could you please add a comment about that in syntax.bnf?

Reply:
I'm not sur it is a good idea, this file is an indication, I don't consider it as normative. Moreover, there are inconsistencies between the lexical conventions in the file and in the parser. For instance, `<ident>` can both include or exclude reserved words.

Reply:
Why not taking syntax.bnf as our reference? Then, a LP parser should implement it.

Reply:
There are many details that you don't want to care about in a high-level document like `syntax.bnf`. It we want to be very precise then it will become less readable I fear.